### PR TITLE
deprecate loop parameter

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#19834

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ class AioHttpTransport(AsyncHttpTransport):
             :caption: Asynchronous transport with aiohttp.
     """
     def __init__(self, *, session: Optional[aiohttp.ClientSession] = None, loop=None, session_owner=True, **kwargs):
+        if loop and sys.version_info >= (3, 10):
+            raise ValueError("Starting Python 3.10, asyncio doesn’t support loop as a parameter anymore")
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:77]: Do we want to add a deprecation warning `if loop and sys.version_info < (3, 10)`?

Comment (annatisch) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:77]: Grammar nitpick: Starting **with** Python 3.10, asyncio no longer supports the loop parameter.

**Review by annatisch — APPROVED** [review]

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:77]: No. If user does not use Python >= 3.10, we want to behave exactly same. There should be no impact at all. (new warnings are still side effects.)

**Pull request merged**